src/containers/CharacterEditForm/CharacterEditForm.js

```diff
@@ -655,7 +657,36 @@ class CharacterEditForm extends React.Component {
     </div>;
   }
 
-  saveTarget() {
+  targets (missedGoals) {
+    if (!missedGoals) return;
+    const targetStatRows =  missedGoals && missedGoals.map((missedGoal, index) =>
```

Comment:
Since `missedGoal` is itself a tuple, it might be nice to break it out into variables so that you don't need to keep indexing it. Something like:

```suggestion
    const targetStatRows =  missedGoals && missedGoals.map(([targetStat, resultValue], index) =>
```

Reply:
Nice yeah done this too.